packages/openapi-parser/src/types/index.ts

```diff
@@ -2,28 +2,47 @@ import type { OpenAPI } from '@scalar/openapi-types'
 
 import type { ERRORS, OpenApiVersion } from '../configuration'
 
+/**
+ * Merge types with each other
+ */
+// TODO: I’d expect merge to overwrite the other way around (overwrite A, keep B)
+export type Merge<A, B> = A & Omit<B, keyof A>
```

Comment:
If the first function returns `version: '2.0' | '3.0' | '3.1'` and the second function returns a more specific `version: '3.1'` it should return the type of the last function (because it can overwrite previous data), but currently it’s returning the return type of the first function. :(